packages/common/src/directives/ng_switch.ts

```diff
@@ -228,7 +233,16 @@ export class NgSwitchCase implements DoCheck {
 export class NgSwitchDefault {
   constructor(
       viewContainer: ViewContainerRef, templateRef: TemplateRef<Object>,
-      @Host() ngSwitch: NgSwitch) {
+      @Optional() @Host() ngSwitch: NgSwitch) {
+    if ((typeof ngDevMode === 'undefined' || ngDevMode) && isDevMode() && !ngSwitch) {
```

Comment:
My understanding is that the `isDevMode()` is not required, since it's essentially a ViewEngine-only version of the `ngDevMode` flag (which is also supported in ViewEngine):
```suggestion
    if ((typeof ngDevMode === 'undefined' || ngDevMode) && !ngSwitch) {
```
The only delta is that the `ngDevMode` would be undefined for apps built with ViewEngine without using Angular CLI (i.e. some custom build setup).

Reply:
@AndrewKushnir you are right. The `isDevMode()` should not be used here as it will break tree-shaking and cause the diagnostic error to be retained in the production code - rendering the `(typeof ngDevMode === 'undefined' || ngDevMode)` check useless.

The only time when `isDevMode()` should be used is for code that we can't tree-shake anyway AND it is too expensive to run that code in production. The only great example of this I can think of right now is the second change detection pass in dev mode to surface `ExpressionChangedAfterItHasBeenCheckedError`.